### ISSUE TITLE
Message Waiting Indicator (NWI) Issue

### ISSUE THREAD

**Issue by andyfor** [fusionpbx/fusionpbx#1193]
Running 4.0.1 Fusion and Freeswitch 1.4.21 and Debian 8.1.

If I leave a voicemail on an extension, the message waiting light comes ON for my Yealink phone, as expected. I do not access the voicemail so the light stays ON.

If I de-register the phone or power cycle the phone or just plug out the ethernet cable and put it back in again, then allow the device to come up and register again, the MWI light is OFF even though a voicemail is still untouched on the system.

I have changed with the MWI subscribe setting(s) on the phone to no avail. Also tried changing the internal sip profile in Fusion, variable send-message-query-on-register to true but again no success.

I can leave another voicemail and the light will then come on and show that this and the old voicemails are in the system so voicemails are fine, just the messaging that seems to have an issue whenever a registration is lost.

Thanks

**Comment by markjcrane** [comment]
If the phone forgets about mwi before its told to remove it then I consider it a bug  in the firmware of the phone.  We do have a script for misbehaving phones in freeswitch/scripts/app/voicemail/resources/scripts an mwi lua script set lua to start this script every time freeswitch starts in the  lua.conf.xml file. It sends reminders to the phone to keep mwi on or off whatever is needed.

**Comment by andyfor** [comment]
Strange though, as it happens on most of the phones I've tested - yealink, grandstream, bria etc.
I added to the startup script but I get this error message:
[ERR] mod_lua.cpp:203 ...eeswitch/scripts/app/voicemail/resources/scripts/mwi.lua:46: /usr/local/freeswitch/scripts/resources/run/voicemail-mwi.tmp: No such file or directory
It's the correct location but there is no "run" directory......

I created a new run directory with www-data permissions and then ran:
fs_cli -x 'luarun app/voicemail/resources/scripts/mwi.lua'

And it all kicked into life. So seems like a fix would be good here. Maybe simplest thing is to remove the /run part as then no need to create a directory.

Thanks
Andy
